langchain4j-open-ai/src/main/java/dev/langchain4j/model/openai/OpenAiStreamingResponseBuilder.java

```diff
@@ -12,19 +12,24 @@
 import dev.langchain4j.model.output.TokenUsage;
 
 import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static dev.langchain4j.model.openai.InternalOpenAiHelper.finishReasonFrom;
 
+/**
+ * This class needs to be thread safe because it is called when a streaming result comes back
+ * and there is no guarantee that this thread will be the same as the one that initiated the request
```

Comment:
Handling thread will be almost definitely **not** the thread that initiated the request, but there should be no problem in this case, or?

Reply:
Exactly, because this class is called by a thread that handles SSE events, it needs to be thread safe.

Reply:
So far I've seen that okhttp handles SSE events using a single thread, and it makes total sense as the order of events is important. So it makes sense that they are consumed sequentially using the same thread. Of course other HTTP frameworks can do this differently, but struggle to see a reason why would they handle a single stream of events using multiple threads, because it would require additional cumbersome synchronization to ensure the order of events preserved. I am not against this change (better safe than sorry), but I am trying to understand the reason.

Reply:
Right, that makes perfect sense!

This comes into play in Quarkus and even Spring Boot if one handles the incoming request in a blocking thread pool, but makes HTTP requests in an async manner (think WebClient of Spring WebFlux) and then waits for the results on the blocking thread pool.

For a user, all that is hidden and there is no synchronization needed.

Reply:
Not sure I understood your response. Do you mean that, when using reactive http clients, this class can receive SSE events from different threads?

Reply:
It can certainly happen yes, because if the client uses an event loop for all IO, then it can very easily happen that the responses come back on a random threads

Reply:
Ok, thanks for explanation!